x-pack/plugins/enterprise_search/public/applications/app_search/components/settings/log_retention/utils/convert_log_retention.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  ELogRetentionOptions,
+  ILogRetention,
+  ILogRetentionPolicy,
+  ILogRetentionServer,
+  ILogRetentionServerPolicy,
+  ILogRetentionServerSettings,
+  ILogRetentionSettings,
+} from '../types';
+
+export const convertLogRetentionFromServerToClient = (
+  logRetention: ILogRetentionServer
+): ILogRetention => ({
+  [ELogRetentionOptions.Analytics]: convertLogRetentionSettingsFromServerToClient(
+    logRetention[ELogRetentionOptions.Analytics]
+  ),
+  [ELogRetentionOptions.API]: convertLogRetentionSettingsFromServerToClient(
+    logRetention[ELogRetentionOptions.API]
+  ),
+});
+
+const convertLogRetentionSettingsFromServerToClient = ({
+  disabled_at: disabledAt,
+  enabled,
+  retention_policy: retentionPolicy,
+}: ILogRetentionServerSettings): ILogRetentionSettings => ({
+  disabledAt,
+  enabled,
+  retentionPolicy:
+    retentionPolicy === null ? null : convertLogRetentionPolicyFromServerToClient(retentionPolicy),
+});
+
+const convertLogRetentionPolicyFromServerToClient = ({
+  min_age_days: minAgeDays,
+  is_default: isDefault,
+}: ILogRetentionServerPolicy): ILogRetentionPolicy => ({
+  isDefault,
+  minAgeDays: minAgeDays === null ? null : minAgeDays,
```

Comment:
I'm pretty confused about this conditional - do we actually need it? It makes sense for `retentionPolicy`, but not so much for minAgeDays. I feel like we could just pass `minAgeDays` straight through here.

Reply:
I don't know, which is why I didn't touch it.

Reply:
https://github.com/elastic/kibana/pull/82364/commits/42cc30b8cb2b4b5b2a8ba402d4c92862889775cf